spring-cloud-dataflow-tasklauncher/spring-cloud-dataflow-tasklauncher-sink/src/main/java/org/springframework/cloud/dataflow/tasklauncher/sink/TaskLauncherSinkConfiguration.java

```diff
@@ -16,84 +16,61 @@
-import org.springframework.cloud.stream.binder.DefaultPollableMessageSource;
-import org.springframework.cloud.stream.binder.PollableMessageSource;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Import;
-import org.springframework.integration.util.DynamicPeriodicTrigger;
 import org.springframework.messaging.Message;
-import org.springframework.messaging.MessageChannel;
-import org.springframework.messaging.MessageHeaders;
-import org.springframework.messaging.support.ChannelInterceptor;
-import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.retry.support.RetryTemplateBuilder;
+import org.springframework.validation.annotation.Validated;
 
 /**
  * Configuration class for the TaskLauncher Data Flow Sink.
  *
  * @author David Turanski
  * @author Gunnar Hillert
  */
-@EnableBinding(PollingSink.class)
-@EnableConfigurationProperties({ TriggerProperties.class })
+@EnableConfigurationProperties({ RetryProperties.class })
 @Import(TaskLauncherFunctionConfiguration.class)
 public class TaskLauncherSinkConfiguration {
 
-	@Value("${autostart:true}")
-	private boolean autoStart;
+	private final static Logger logger = LoggerFactory.getLogger(TaskLauncherSinkConfiguration.class);
+
+	static class LaunchRequestConsumer implements LaunchRequestMessageConsumer {
+
+		private final TaskLauncherFunction taskLauncherFunction;
+
+		public LaunchRequestConsumer(TaskLauncherFunction taskLauncherFunction) {
+			this.taskLauncherFunction = taskLauncherFunction;
+		}
+
+		@Override
+		public void accept(Message<LaunchRequest> message) {
+			taskLauncherFunction.accept(message.getPayload());
+		}
 
-	@Bean
-	public DynamicPeriodicTrigger periodicTrigger(TriggerProperties triggerProperties) {
-		DynamicPeriodicTrigger trigger = new DynamicPeriodicTrigger(triggerProperties.getPeriod());
-		trigger.setInitialDuration(Duration.ofMillis(triggerProperties.getInitialDelay()));
-		return trigger;
 	}
 
-	/*
-	 * For backward compatibility with spring-cloud-stream-2.1.x
-	 */
-	@Bean
-	public BeanPostProcessor addInterceptorToPollableMessageSource() {
-		return new BeanPostProcessor() {
-			@Override
-			public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
-				if (bean instanceof DefaultPollableMessageSource) {
-					DefaultPollableMessageSource pollableMessageSource = (DefaultPollableMessageSource) bean;
-					pollableMessageSource.addInterceptor(new ChannelInterceptor() {
-						@Override
-						public Message<?> preSend(Message<?> message, MessageChannel channel) {
-							Message<?> newMessage = message;
-							if (message.getHeaders().containsKey("originalContentType")) {
-								newMessage = MessageBuilder.fromMessage(message)
-										.setHeader(MessageHeaders.CONTENT_TYPE,
-												message.getHeaders().get("originalContentType"))
```

Comment:
Is this all magically handled by SI now?

Reply:
We aren't using an explicit message source. 
The message is processed as JSON for LaunchRequest.